lib/storage/index_db.go

```diff
@@ -224,16 +232,26 @@ func (db *indexDB) UpdateMetrics(m *IndexDBMetrics) {
 
 	m.MissingMetricNamesForMetricID += db.missingMetricNamesForMetricID.Load()
 
-	m.IndexBlocksWithMetricIDsProcessed = indexBlocksWithMetricIDsProcessed.Load()
-	m.IndexBlocksWithMetricIDsIncorrectOrder = indexBlocksWithMetricIDsIncorrectOrder.Load()
-
-	m.MinTimestampForCompositeIndex = uint64(db.s.minTimestampForCompositeIndex)
-	m.CompositeFilterSuccessConversions = compositeFilterSuccessConversions.Load()
-	m.CompositeFilterMissingConversions = compositeFilterMissingConversions.Load()
-
 	db.tb.UpdateMetrics(&m.TableMetrics)
 	db.doExtDB(func(extDB *indexDB) {
 		extDB.tb.UpdateMetrics(&m.TableMetrics)
+
+		cs.Reset()
+		extDB.tagFiltersToMetricIDsCache.UpdateStats(&cs)
+		m.TagFiltersToMetricIDsCacheSize += cs.EntriesCount
+		m.TagFiltersToMetricIDsCacheSizeBytes += cs.BytesSize
+		m.TagFiltersToMetricIDsCacheSizeMaxBytes += cs.MaxBytesSize
+		m.TagFiltersToMetricIDsCacheRequests += cs.GetCalls
+		m.TagFiltersToMetricIDsCacheMisses += cs.Misses
+
+		m.IndexDBRefCount += uint64(extDB.refCount.Load())
+		m.MissingTSIDsForMetricID += extDB.missingTSIDsForMetricID.Load()
+
+		m.DateRangeSearchCalls += extDB.dateRangeSearchCalls.Load()
+		m.DateRangeSearchHits += extDB.dateRangeSearchHits.Load()
+		m.GlobalSearchCalls += extDB.globalSearchCalls.Load()
+
+		m.MissingMetricNamesForMetricID += extDB.missingMetricNamesForMetricID.Load()
```

Comment:
WDYT about moving this to a function of `indexDB` similar to `mergeset.Table.UpdateStats` / `workingsetcache.Cache.UpdateStats`?
Seems like this will enforce metrics to be updated for both old and current `indexDB` in a consistent way.

Reply:
I think it requires additional refactoring. IndexDB exposes global and local metrics. It'd be great to unify it, but it's not an easy task with current implementation. 

 Probably, I can add additional function for local metrics and use it both indexdbs